### ISSUE TITLE
Can I get processed audio from the KaldiRecognizer?

### ISSUE THREAD

**Issue by chamecall** [alphacep/kaldi-android-demo#16]
I had to refuse android.SpeechRecognizer because of the impossibility to extract audio from it so I'd like to find out how I can reach that purpose in here..

**Comment by nshmyrev** [comment]
If you record audio yourself you can simply feed it into the recognizer, see https://github.com/alphacep/kaldi-android-demo/blob/83b23cfc34dde4b688fbfff7566a460bc9b56455/app/src/main/java/org/kaldi/demo/KaldiActivity.java#L132 method

**Comment by abouquet** [comment]
I guess he wants to do the opposite : extract recognized audio and process it for an other use case (like saving audio AND text together)

**Comment by antondim** [comment]
would it be helpful to add such a feature in vosk-api python version?

**Comment by pristalovpavel** [comment]
Are there any updates with this issue? Or maybe someone else added this function?

**Comment by l3von** [comment]
Are there any updates on this issue. Is it possible to get audio file from the default recognizer used by Vosk or no?

**Comment by nanaghartey** [comment]
> Are there any updates on this issue. Is it possible to get audio file from the default recognizer used by Vosk or no?

Yes, it's possible. You can read in raw PCM data from android's AudioRecord and output it to a valid WAV file. You'd have to implement in SpeechService class and build..

**Comment by 7Hrant7** [comment]
> > Are there any updates on this issue. Is it possible to get audio file from the default recognizer used by Vosk or no?
> 
> Yes, it's possible. You can read in raw PCM data from android's AudioRecord and output it to a valid WAV file. You'd have to implement in SpeechService class and build..

Can you send example?

**Comment by l3vonPreezma** [comment]
Did you create your own implementation of SpeechService? Because inside original implementation AudioRecord  field is private.

**Comment by nanaghartey** [comment]
> Did you create your own implementation of SpeechService? Because inside original implementation AudioRecord field is private.

Yes I did!

**Comment by PavelPristalovArammeem** [comment]
@nanaghartey could you share your fork please? I think the whole community will be grateful to you for this.
